### ISSUE TITLE
aws-batch-alpha: default instance role created by ManagedEc2EcsComputeEnvironment doesn't have correct permissions

### ISSUE THREAD

**Issue by damonmaria** [aws/aws-cdk#25305]
### Describe the bug

The default `instanceRole` created by `ManagedEc2EcsComputeEnvironment` does not have the required permissions / policies. I get the following in ClourTrail using the default role:
```
User: arn:aws:sts::***:assumed-role/model-etl-gpuModelEtlComputeEnvInstanceProfileRole-1LCGMU0AF61LQ/i-03e81913e0cb8a0f2 is not authorized to perform: ecs:RegisterContainerInstance on resource: arn:aws:ecs:us-east-1:***:cluster/AWSBatch-gpuModelEtlComputeEnv775-FgPkH7AlQz6YeGud-fc1bce46-cf4f-3cc9-9529-903f0c428dba because no identity-based policy allows the ecs:RegisterContainerInstance action
```

### Expected Behavior

The default instance role create by `ManagedEc2EcsComputeEnvironment` can run containers. 

I expect it needs the managed policy `service-role/AmazonEC2SpotFleetTaggingRole` attached? 

### Current Behavior

Default role has no permissions at all and so fails to `RegisterContainerInstance` with above error.

### Reproduction Steps

Try and run a batch job in an `ManagedEc2EcsComputeEnvironment` using the default instance role. 

### Possible Solution

If I attach the managed policy `service-role/AmazonEC2SpotFleetTaggingRole` to the generated default instance role it works. 

### CDK CLI Version

@aws-cdk/aws-batch-alpha: 2.76.0-alpha.0, aws-cdk: 2.76.0 

### Node.js Version

v18.15.0

### OS

MacOS 13.3.1

### Language

Typescript

### Language Version

TypeScript 5.0.4

**Comment by pahud** [comment]
Looks like the default generated role does not have any managed policies.

https://github.com/aws/aws-cdk/blob/046d0fce6904521eb4d22bf0fb69afe5cef139be/packages/%40aws-cdk/aws-batch-alpha/lib/managed-compute-environment.ts#L641

Can you share your CDK sample code so we can see more details about it?

**Comment by TheRealAmazonKendra** [comment]
Looks like this is a duplicate of https://github.com/aws/aws-cdk/issues/25305

**Comment by damonmaria** [comment]
> Looks like this is a duplicate of #25305

@TheRealAmazonKendra Did you mean to link to a different issue? Issue #25305 is this one.

**Comment by TheRealAmazonKendra** [comment]
> > Looks like this is a duplicate of #25305
> 
> @TheRealAmazonKendra Did you mean to link to a different issue? Issue #25305 is this one.

Yes, yes I did.

https://github.com/aws/aws-cdk/issues/25256

**Comment by TheRealAmazonKendra** [comment]
Did the linked fix in that issue resolve this one?

**Comment by damonmaria** [comment]
> Did the linked fix in that issue resolve this one?

Ah, yes. That is actually what I am doing already. I copy and pasted the wrong managed policy name. `service-role/AmazonEC2ContainerServiceforEC2Role` is the right one. 

Closing this is a duplicate #25256. I did search in Github before composing this issue but finished submitting the issue the next day. In the meantime #25256 must have been posted.
